platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslBaseScriptModelBuilder.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.tooling.builders
+
+import org.gradle.internal.build.BuildState
+import org.gradle.kotlin.dsl.provider.KotlinScriptClassPathProvider
+import org.gradle.kotlin.dsl.support.ImplicitImports
+import org.gradle.kotlin.dsl.support.serviceOf
+import org.gradle.tooling.model.kotlin.dsl.KotlinDslBaseScriptModel
+import org.gradle.tooling.provider.model.internal.BuildScopeModelBuilder
+import java.io.File
+import java.io.Serializable
+
+
+internal
+object KotlinDslBaseScriptModelBuilder : BuildScopeModelBuilder {
+    override fun canBuild(modelName: String): Boolean =
+        modelName == "org.gradle.tooling.model.kotlin.dsl.KotlinDslBaseScriptModel"
+
+    override fun create(target: BuildState): KotlinDslBaseScriptModel {
+        val implicitImports = target.getMutableModel().serviceOf<ImplicitImports>()
+        val kotlinScriptClassPathProvider = target.getMutableModel().serviceOf<KotlinScriptClassPathProvider>()
+        return StandardKotlinDslBaseScriptModel(
+            implicitImports = implicitImports.list,
+            kotlinDslClassPath = kotlinScriptClassPathProvider.gradleKotlinDsl.asFiles
+        )
+    }
+}
+
+
+internal
+data class StandardKotlinDslBaseScriptModel(
```

Comment:
💅  Why not `Default` prefix for an implementation class that we usually use?

Reply:
I used the same names as they are used for other models, e.g.:
https://github.com/gradle/gradle/blob/6e6fd9a24268d197a8c14c1165daa124d2c346ef/platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt#L37

Reply:
@bamboo, @eskatos, any reason why we shouldn't unify the implementation class naming for Kotlin models with the rest of Gradle?

It's tiny friction when things are not uniform (e.g. when searching in the IDE), and we have a lot of those. I personally appreciate unification over diversity when it comes to maintaining Gradle codebase.